source/dem/grid_motion.cc

```diff
@@ -51,6 +79,96 @@ GridMotion<dim>::move_grid_translational(
   GridTools::shift(shift_vector, triangulation);
 }
 
+template <int dim>
+void
+GridMotion<dim>::move_grid_due_particles_forces(
+  parallel::distributed::Triangulation<dim> &triangulation)
+{
+  update_parameters_before_motion();
+  calculate_motion_parameters();
+  move_grid_translational(triangulation);
+  // TODO : insert a move_grid_rotational(triangulation); The problem right now
+  //  is that this function make the rotation around (0,0,0) point and not
+  //  around the center of mass, which is moving
+  update_parameters_after_motion();
+}
+
+template <int dim>
+void
+GridMotion<dim>::calculate_motion_parameters()
+{
+  unsigned int this_mpi_process(
+    Utilities::MPI::this_mpi_process(MPI_COMM_WORLD));
+
+  if (this_mpi_process == 0)
+    {
+      Tensor<1, dim> translational_velocity_one_step_time_further;
+      Tensor<1, 3>   rotational_velocity_one_step_time_further;
+
+      translational_velocity_one_step_time_further =
+        (dt / triangulation_mass) * triangulation_forces +
```

Comment:
what if triangulation_mass or intertia are zero? you need to throw a specific error for these conditions

Reply:
Yes that's really good point. I let the value be nan for [0] and [2] vector coordinate. I fixed the inertia tensor to be three of length and as default (0,0,0), maybe just a message telling that information in x or y or z axis is wrong but not throwing an error. Like i still wanna do it even with zero information for inertia (for full zero, we throw an error). But for triangulation_mass, if it's zero, yes we should throw an error. Good point again.